packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/CameraPlugin.m

```diff
@@ -117,14 +137,13 @@ - (void)availableCamerasWithCompletion:
     if (@available(iOS 13.0, *)) {
       [discoveryDevices addObject:AVCaptureDeviceTypeBuiltInUltraWideCamera];
     }
-    AVCaptureDeviceDiscoverySession *discoverySession = [AVCaptureDeviceDiscoverySession
-        discoverySessionWithDeviceTypes:discoveryDevices
-                              mediaType:AVMediaTypeVideo
-                               position:AVCaptureDevicePositionUnspecified];
-    NSArray<AVCaptureDevice *> *devices = discoverySession.devices;
+    NSArray<id<FLTCaptureDevice>> *devices =
+        [self.deviceDiscoverer discoverySessionWithDeviceTypes:discoveryDevices
+                                                     mediaType:AVMediaTypeVideo
+                                                      position:AVCaptureDevicePositionUnspecified];
     NSMutableArray<FCPPlatformCameraDescription *> *reply =
         [[NSMutableArray alloc] initWithCapacity:devices.count];
-    for (AVCaptureDevice *device in devices) {
+    for (id<FLTCaptureDevice> device in devices) {
```

Comment:
Why is this cast necessary? This is how the array was typed above.

Reply:
It's not really a cast, we have to provide the type in Obj-C for..in loops, so this is the exact type of the array item as the array is declared as `NSArray<id<FLTCaptureDevice>> *devices`. Previously that loop was operating on `AVCaptureDevice*`, but now that the session discovery is abstracted away this operates on the new protocol (it only accesses `uniqueID` and `position` properties inside).

Reply:
> It's not really a cast, we have to provide the type in Obj-C for..in loops

Sorry, I have no idea what happened to my reading comprehension when I was reviewing that 🤦🏻